Bottom: 0fea11af3a860d1ff70ab2ee05e97fdb7afa2ed1
Top:    d386bc5cbc520954e64a946513b18a251407c89e
Author: Seva Gribov <seva.gribov@aerq.com>
Date:   2021-02-11 15:16:25 +0000

Refresh of scroll-back-through-terminal

---

diff --git a/st.c b/st.c
index e2dd722..8d0e0ad 100644
--- a/st.c
+++ b/st.c
@@ -1074,7 +1074,7 @@ kscrolldown(const Arg* a)
 	int n = a->i;
 
 	if (n < 0)
-		n = term.row + n;
+		n = term.row/2 + n;
 
 	if (n > term.scr)
 		n = term.scr;
@@ -1092,7 +1092,7 @@ kscrollup(const Arg* a)
 	int n = a->i;
 
 	if (n < 0)
-		n = term.row + n;
+		n = term.row/2 + n;
 
 	if (term.scr <= HISTSIZE-n) {
 		term.scr += n;
